util/topgen/merge.py

```diff
@@ -747,6 +747,52 @@ def amend_resets(top, name_to_block):
     top["resets"] = top_resets
 
 
+def create_alert_lpgs(top, name_to_block: Dict[str, IpBlock]):
+    '''Loop over modules and determine number of unique LPGs'''
+    num_lpg = 0
+    lpg_dict = {}
+    top['alert_lpgs'] = []
+    for module in top["module"]:
+        # the alert senders are attached to the primary clock of this block,
+        # so let's start by getting that primary clock.
+        block = name_to_block[module['type']]
+        block_clock = block.get_primary_clock()
```

Comment:
hmm... does this work? because this gets you what the primary clock is in the hjson but not the top level connection right..?

Reply:
see next comment.